Metalama.Framework.Engine/Linking/LinkerInjectionStep.TransformationCollection.cs

```diff
@@ -0,0 +1,344 @@
+        }
+
+        public MemberLevelTransformations GetOrAddMemberLevelTransformations( IDeclarationBuilder declarationBuilder )
+        {
+            return this._introductionMemberLevelTransformations.GetOrAdd( declarationBuilder, static _ => new MemberLevelTransformations() );
+        }
+
+        internal IReadOnlyList<StatementSyntax> GetInjectedInitializerStatements( InjectedMember injectedMember )
+        {
+            if ( injectedMember.Declaration is not IMember member
+                 || !this._insertedStatementsByTargetDeclaration.TryGetValue( member, out var insertedStatements ) )
+            {
+                return ImmutableArray<StatementSyntax>.Empty;
+            }
+
+            var statements = new List<StatementSyntax>();
+
+            if ( this._injectedMembersByTargetDeclaration.TryGetValue(member, out var injectedMembers ) 
+                 && injectedMembers[^1] == injectedMember )
+            {
+                var initializerStatements = 
+                    insertedStatements
+                    .Where( s => s.InsertedStatement.Kind == InsertedStatementKind.Initializer )
+                    .Select( s => s.InsertedStatement );
+
+                var orderedInitializerStatements = Order( initializerStatements );
+
+                // This is the last override, insert all Initializer statements.
+                statements.AddRange( orderedInitializerStatements.Select( s => s.Statement ) );
+            }
+
+            return statements;
+
+            static IEnumerable<InsertedStatement> Order( IEnumerable<InsertedStatement> statements )
+                => statements
+                    .OrderBy(
+                        s => s.ContextDeclaration switch
+                        {
+                            IMember => 0,
+                            INamedType => 1,
+                            _ => throw new AssertionFailedException( $"Unexpected declaration: '{s.ContextDeclaration}'." )
+                        } )
+                    .ThenBy( s => (s.ContextDeclaration as IMember)?.ToDisplayString() );
+        }
+
+        internal IReadOnlyList<StatementSyntax> GetInjectedEntryStatements( IMember sourceMember )
+        {
+            if ( !this._insertedStatementsByTargetDeclaration.TryGetValue( sourceMember, out var insertedStatements ) )
+            {
+                return ImmutableArray<StatementSyntax>.Empty;
+            }
+
+            var statements = new List<StatementSyntax>();
+
+            if ( !this._injectedMembersByTargetDeclaration.TryGetValue( sourceMember, out var injectedMembers )
+                || injectedMembers.Count == 0)
+            {
+                // This is the last override, insert all Initializer statements.
+                statements.AddRange( insertedStatements.Where( s => s.InsertedStatement.Kind == InsertedStatementKind.Initializer ).Select( s => s.InsertedStatement.Statement ) );
+            }
+
+            return statements;
```

Comment:
Why is ordering not necessary in this case? If it's the correct behavior, maybe worth adding a comment about it?